web_full_stack/js/src/ListingsPage.js

```diff
@@ -1,34 +1,28 @@
-import React, { Component } from 'react';
+import React, { useEffect, useState } from 'react';
 import { fetchListings } from './API';
 
-export default class ListingsPage extends Component {
-  state = {
-    listings: [],
-  }
+export default function ListingsPage() {
+  const [listings, setListings] = useState([]);
 
-  async componentDidMount() {
-    const resp = await fetchListings();
-
-    this.setState({
-      listings: resp.listings,
+  useEffect(() => {
```

Comment:
Later they'll need to add pieces of the react router props to the deps list right? They have to filter eventually?

Reply:
That's right